src/Features/TransientNotices.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * WooCommerce Transient Notices
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+use \Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Shows print shipping label banner on edit order page.
+ */
+class TransientNotices {
+
+	/**
+	 * Option name for the queue.
+	 */
+	const QUEUE_OPTION = 'woocommerce_admin_transient_notices_queue';
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_notices' ) );
+	}
+
+	/**
+	 * Add a notice to be shown.
+	 *
+	 * @param array $notice Notice.
+	 *    $notice = array(
+	 *      'id'      => (string) Unique ID for the notice. Required.
+	 *      'status'  => (string) info|error|success
+	 *      'content' => (string) Content to be shown for the notice. Required.
+	 *      'options' => (array) Array of options to be passed to the notice component.
+	 *       See https://developer.wordpress.org/block-editor/reference-guides/data/data-core-notices/#createNotice for available options.
+	 *    ).
+	 */
+	public static function add( $notice ) {
+		$queue = get_option( self::QUEUE_OPTION, array() );
```

Comment:
How does this work with multiple admin users?

For a while I've suspected there's a bug with the CES queue where users might see notices generated by another user if they are both active in the admin a the same time.

I wonder if this should be stored in user metadata instead?

Reply:
That's a good question.  I suspect this is a little edge case in terms of timing, but certainly a bug that could happen and probably a bug that does exist among many similar actions that are dependent on the next page load.  

I'm thinking if we do this, we could handle in a follow-up.  Do you think making notifications user specific are worthwhile?  Same question in regards to page specificity (only display on certain pages)?

Reply:
Yeah a follow-up sounds good 👍 

> Do you think making notifications user specific are worthwhile?

I think it would be nice to have. For CES at least, a user needs to dismiss the notice or interact with it for it to leave the queue. As such, if they do nothing then the notice stays in the queue and then I suspect other users could see it on their next page view depending on the timing. 

It's still just a theory I have so I'm happy to investigate it some more. I would just like to avoid creating any confusion where a user get a notice that is out of context for them.

> Same question in regards to page specificity (only display on certain pages)?

This might be a CES-specific thing that wouldn't be needed normally. It was put in place to make sure notices only get triggered from certain pages. For example, the product notice only shows after updating the product from the product edit page. We wanted to avoid showing a notice when the product was updated elsewhere, like WP CLI, importer, bulk-editing etc.

Reply:
I agree that this would cause notices to bleed through to other users. IMO making it per-user is worthwhile. Customer Effort Score also had the requirement that the notice only be displayed once per _store_ rather than once per _user_, which is most likely why I was ok with saving it as an option rather than per-user.

Reply:
I went ahead and added a `user_id` option to notices.  Any notices with a `user_id` will be shown to that user.  Notices without any `user_id` can be viewed by any user.

Example with the `PluginsInstaller` updated to show this in action.

Reply:
Thanks and nice job on this @joshuatf!